.github/workflows/ensure_linting.yml

```diff
@@ -0,0 +1,23 @@
+name: Ensure Linting
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  check_linting:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Install Nix
+        uses: cachix/install-nix-action@ba0dd844c9180cbf77aa72a116d6fbc515d0e87b #v27
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+          github_access_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: run treefmt in CI mode
+        run: nix flake check --impure
```

Comment:
Just want to call out that `nix flake check` will run *all* checks (which is something we really probably should be doing in CI). IMO we should just rename this job to something like "run nix checks" (and adjust any comments as needed) and call it a day.

If you do specifically want to just run the formatter here, unfortunately there's no built-in way to run a specific check so you'd have to do something like `nix build .#checks.x86_64-linux.formatting` (or just invoke treefmt outside of nix entirely)

Reply:
> Just want to call out that `nix flake check` will run _all_ checks (which is something we really probably should be doing in CI). IMO we should just rename this job to something like "run nix checks" (and adjust any comments as needed) and call it a day.

Yeah, will add this to CI as well, which enhance our elixir test cases with a real test of db connection for example.


> If you do specifically want to just run the formatter here, unfortunately there's no built-in way to run a specific check so you'd have to do something like `nix build .#checks.x86_64-linux.formatting` (or just invoke treefmt outside of nix entirely)

Reply:
> Yeah, will add this to CI as well, which enhance our elixir test cases with a real test of db connection for example.

will be tracked with #4234